src/modules/cmdpal/Microsoft.CmdPal.UI/LoadIconBehavior.xaml.cs

```diff
@@ -14,32 +14,36 @@ public partial class LoadIconBehavior : DependencyObject, IBehavior
 {
     private static readonly IconCacheService IconService = new(Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread());
 
-    public IconDataType Source
+    public IconDataType? Source
     {
-        get => (IconDataType)GetValue(SourceProperty);
-        set
-        {
-            SetValue(SourceProperty, value);
-            OnSourcePropertyChanged();
-        }
+        get => (IconDataType?)GetValue(SourceProperty);
+        set => SetValue(SourceProperty, value);
     }
 
     // Using a DependencyProperty as the backing store for Source.  This enables animation, styling, binding, etc...
     public static readonly DependencyProperty SourceProperty =
-        DependencyProperty.Register(nameof(Source), typeof(IconDataType), typeof(LoadIconBehavior), new PropertyMetadata(new IconDataType(string.Empty)));
+        DependencyProperty.Register(nameof(Source), typeof(IconDataType), typeof(LoadIconBehavior), new PropertyMetadata(null, OnSourcePropertyChanged));
 
     public DependencyObject? AssociatedObject { get; private set; }
 
     public void Attach(DependencyObject associatedObject) => AssociatedObject = associatedObject;
 
     public void Detach() => AssociatedObject = null;
 
-    public async void OnSourcePropertyChanged()
+    private static async void OnSourcePropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
     {
-        var icoSource = await IconService.GetIconSource(Source ?? new(string.Empty));
-
-        if (AssociatedObject is Border border)
+        if (d is LoadIconBehavior @this
+            && @this.AssociatedObject is Border border)
         {
+            var icoSource = await IconService.GetIconSource(@this.Source ?? new(string.Empty));
```

Comment:
I was so excited about this behavior too. It's always too good to be true.

RE: #213: 

I may explore here if we need to have a cancellation here, or some sort of mutex per instance. Basically, I think a new request is coming in and then when we get here it's not guaranteed which one finishes, but we want to ignore the originals and ensure we only finish the last one.